packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts

```diff
@@ -0,0 +1,160 @@
+import debounce from 'lodash.debounce'
+import { rawLayerList } from '@masterportal/masterportalapi'
+import {
+  Feature as GeoJsonFeature,
+  GeoJsonProperties,
+  Geometry as GeoJsonGeometry,
+} from 'geojson'
+import { MapConfig, PolarActionContext } from '@polar/lib-custom-types'
+import { Map, Feature } from 'ol'
+import { Geometry } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import { addFeature } from '../../utils/displayFeatureLayer'
+import { requestGfi } from '../../utils/requestGfi'
+import sortFeatures from '../../utils/sortFeatures'
+import { GfiGetters, GfiState } from '../../types'
+
+const mapFeaturesToLayerIds = (
+  layerKeys: string[],
+  configuration: MapConfig,
```

Comment:
I still prefer if this would be `GfiConfiguration` (`configuration.gfi`), as the two instances could be shortened and the second optional chaining would be redundant.

Reply:
[I https://github.com/Dataport/polar/pull/](https://github.com/Dataport/polar/pull/176/commits/6ab8b0d1491c22bc26f6709199344772f3b49c73)